### PR TITLE
Remove warnings and improve colorbar gallery example

### LINKED ISSUE #595: Improve the colorbar example
**Description of the problem**

We already have an example for colorbar in the gallery (https://www.pygmt.org/dev/gallery/plot/colorbar.html). However, the example is a little low-quality.

1. The example gives many warnings like:
   ```
   colorbar [WARNING]: Axis sub-item f set more than once (typo?)
   colorbar [WARNING]: Axis sub-item a set more than once (typo?)
   ```

2. The example is not beautiful at all. The main reason is that the figure size is too small and the colorbars are too large.
3. The example is not as usable as I expect. I believe very few people will plot colorbars inside a frame. IMHO, at least we should show
     1. calling `colorbar` with very few arguments, using the default location and default size
     2. the easiest way (using **J**) to plot colorbars outside the basemap.

@weiji14 As the example was created by you, I'd like to hear what you think. You can improve the example if you like.

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Prevents the "Axis sub-item a/f set more than once (typo?)" warning from showing up, and also make the `colorbar` gallery example a bit nicer than the original one made in #332.

Live preview is at https://pygmt-git-improve-gallery-colorbar.gmt.vercel.app/gallery/plot/colorbar.html. See also teaser image below (which should dynamically update if the code in this PR changes).

New colorbar gallery example

Fixes #595

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: examples/gallery/plot/colorbar.py [hunk 1]
```diff
@@ -2,51 +2,59 @@
...
 import pygmt
 
 fig = pygmt.Figure()
-fig.basemap(region=[0, 3, 6, 9], projection="t0/3c", frame=True)
+fig.basemap(region=[0, 3, 6, 9], projection="x3c", frame=["af", 'WSne+t"Colorbars"'])
 
-# Create a colorbar suitable for surface topography- oleron
+## Create a colorbar designed for seismic tomography- roma
+# Colorbar is placed at bottom center (BC) by default if no position is given
+fig.colorbar(cmap="roma", frame=["+Lvelocity", "y+lm/s"])
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:33]: Why are you using `+Lvelocity`? `x+lvelocity` make more sense to me.
```suggestion
fig.colorbar(cmap="roma", frame=["x+lvelocity", "y+lm/s"])
```

File: examples/gallery/plot/colorbar.py [hunk 2]
```diff
@@ -2,51 +2,59 @@
...
-You can set the `position` of the colorbar using the following options:
-
-- j/J: justified inside/outside the mapframe using any 2 character combination of
-  vertical (**T** op, **M** iddle, **B** ottom) and horizontal (**L** eft, **C** enter,
-  **R** ight) alignment codes, e.g. `position="jTR"` for top right.
-- g: using map coordinates, e.g. `position="g170/-45"` for longitude 170E, latitude 45S.
-- x: using paper coordinates, e.g. `position="x5c/7c"` for 5cm,7cm from anchor point.
+The :meth:`pygmt.Figure.colorbar` method creates a color scalebar. We must
+specify the colormap via the ``cmap`` argument, and set the placement via the
+``position`` argument. The full list of color palette tables can be found at
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:7]: `position` is not required.

File: examples/gallery/plot/colorbar.py [hunk 3]
```diff
@@ -2,51 +2,59 @@
...
+:gmt-docs:`cookbook/cpts.html`. You can set the `position` of the colorbar
+using the following options:
+
+- j/J: justified inside/outside the mapframe using any 2 character combination
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:11]:
```suggestion
- j/J: justified inside/outside the map frame using any 2 character combination
```

File: examples/gallery/plot/colorbar.py [hunk 4]
```diff
@@ -2,51 +2,59 @@
...
+## Create a colorbar designed for seismic tomography- roma
+# Colorbar is placed at bottom center (BC) by default if no position is given
+fig.colorbar(cmap="roma", frame=["+Lvelocity", "y+lm/s"])
 
+## Create a colorbar showing the scientific rainbow - batlow
 fig.colorbar(
-    cmap="oleron",
-    position="jTC+w6c/1c+h",  # justified inside map frame (j) at Top Center (TC)
+    cmap="batlow",
+    # Colorbar positioned at map coordinates (g) longitude/latitude 0.2/8.7,
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:38]:
```suggestion
    # Colorbar positioned at map coordinates (g) longitude/latitude 0.3/8.7,
```

File: examples/gallery/plot/colorbar.py [hunk 5]
```diff
@@ -2,51 +2,59 @@
...
+    # with a length/width (+w) of 4cm by 0.5cm, and plotted horizontally (+h)
+    position="g0.3/8.7+w4c/0.5c+h",
     box=True,
-    frame=["+Loleron", "xaf", "y+lm"],
-    scale=10,
+    frame=["+Ltemperature", r"y+l\260C"],
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:42]:
```suggestion
    frame=["x+ltemperature", r"y+l\260C"],
```

File: examples/gallery/plot/colorbar.py [hunk 6]
```diff
@@ -2,51 +2,59 @@
...
-    box=True,
-    frame=["+Lbatlow", "xaf", r"y+l\260C"],
+    cmap="oleron",
+    # Colorbar position justified outside map frame (J) at Middle Right (MR),
+    # offset (+o) by 1cm horizontally and 0cm vertically from anchor point,
+    # with a length/width (+w) of 7cm by 0.5cm and a box for NaN values (+n)
+    position="JMR+o1c/0c+w7c/0.5c+n+mc",
+    # The label (+L) 'Elevation' is moved to opposite side and plotted
+    # vertically as a column of text using (+mc) in the position argument above
+    frame=["+LElevation", "y+lm"],
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:55]:
```suggestion
    frame=["x+lElevation", "y+lm"],
```

File: examples/gallery/plot/colorbar.py [hunk 7]
```diff
@@ -2,51 +2,59 @@
...
     scale=10,
 )
 
+
```
Comment (seisman) [inline examples/gallery/plot/colorbar.py:59]:
```suggestion
```

File: examples/gallery/plot/colorbar.py (hunk 1 above)
Comment (weiji14) [reply examples/gallery/plot/colorbar.py:33]: Hmm you're right, must have got it from some old GMT example before.

**Review by seisman — APPROVED** [review]

**Pull request merged**